api/types/src/view.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{HexEncodedBytes, MoveType};
+use poem_openapi::Object;
+use serde::{Deserialize, Serialize};
+
+/// Table Item request for the GetTableItem API
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Object)]
+pub struct ViewRequest {
+    pub arguments: Vec<HexEncodedBytes>,
```

Comment:
right now the input argument is just a bcs serialized bytes for input arguments. Maybe we want to use `(MoveValue, MoveType)` instead?

Reply:
Isn't that anyway only the raw REST API level? I thought via our api spec we are generating bindings on top of this? 
@banool to shed some light on this 

Also there are those typescript APIs which use the Move abis -- how are those related to this?

Reply:
I would take whatever comes from `EntryFunctionPayload` and do the exact same thing here.
```
* args: Vec<HexEncodedBytes>
* type_args: Vec<MoveType>
```

The bindings built on top are outside the scope of this and built only for typescript